pyo3-macros-backend/src/pyclass.rs

```diff
@@ -689,21 +707,28 @@ fn impl_simple_enum(
     let variants = simple_enum.variants;
     let pytypeinfo = impl_pytypeinfo(cls, args, None, ctx);
 
+    for variant in &variants {
+        ensure_spanned!(variant.options.constructor.is_none(), variant.options.constructor.span() => "`constructor` can't be used on a simple enum variant");
+    }
+
     let (default_repr, default_repr_slot) = {
-        let variants_repr = variants.iter().map(|variant| {
-            let variant_name = variant.ident;
-            // Assuming all variants are unit variants because they are the only type we support.
-            let repr = format!(
-                "{}.{}",
-                get_class_python_name(cls, args),
-                variant.get_python_name(args),
-            );
-            quote! { #cls::#variant_name => #repr, }
-        });
+        let variants_repr = variants
+            .iter()
+            .map(|variant| {
+                let variant_name = variant.ident;
+                // Assuming all variants are unit variants because they are the only type we support.
+                let repr = format!(
+                    "{}.{}",
+                    get_class_python_name(cls, args),
+                    variant.get_python_name(args),
+                );
+                Ok(quote! { #cls::#variant_name => #repr, })
+            })
+            .collect::<Result<TokenStream>>()?;
```

Comment:
I think this collect can be made infallible again.

Reply:
Ah, yes. I reverted it to how it was before 👍